internal/precache/helper.go

```diff
@@ -36,16 +39,14 @@ import (
 	"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
-func PathOutsideChroot(filename string) string {
-	return filepath.Join(utils.Host, filename)
-}
-
+// getConfigMap retrieves the specified ConfigMap via common.GetConfigMap function, if the ConfigMap is not found,
+// nil is returned
 func getConfigMap(ctx context.Context, c client.Client, name, namespace string) (*corev1.ConfigMap, error) {
-	cm := &corev1.ConfigMap{}
-	if err := c.Get(ctx, types.NamespacedName{
+	cm, err := common.GetConfigMap(ctx, c, v1alpha1.ConfigMapRef{
 		Name:      name,
 		Namespace: namespace,
-	}, cm); err != nil {
+	})
+	if err != nil {
```

Comment:
Actually, do you still need this helper function in precache? You could directly call the util one in the validateJobConfig and only add the IsNotFound check?

Reply:
Thanks Angie, I have refactored as per you suggestion.